nautobot_device_lifecycle_mgmt/filters.py

```diff
@@ -136,6 +137,132 @@ def _expired_search(self, queryset, name, value):  # pylint: disable=unused-argu
         return queryset.filter(qs_filter)
 
 
+class SoftwareNoticeFilterSet(NautobotFilterSet):
+    """Filter for SoftwareNotice."""
+
+    q = SearchFilter(
+        filter_predicates={
+            "software_version__version": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "software_version__alias": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "device_type__model": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "device_type__part_number": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "comments": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "documentation_url": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "end_of_sale": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "end_of_support": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "end_of_sw_releases": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "end_of_security_patches": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+        }
+    )
+    software_version_id = django_filters.ModelMultipleChoiceFilter(
+        field_name="software_version",
+        queryset=SoftwareVersion.objects.all(),
+        label="Software Version",
+    )
+    software_version = django_filters.ModelMultipleChoiceFilter(
+        field_name="software_version", queryset=SoftwareVersion.objects.all(), label="Software Version"
+    )
+    software_version_version = django_filters.ModelMultipleChoiceFilter(
+        field_name="software_version__version",
+        queryset=SoftwareVersion.objects.all(),
+        to_field_name="version",
+        label="Software Version (String)",
+    )
```

Comment:
Can/should these three be consolidated into a single filter using `NaturalKeyOrPKMultipleChoiceFilter`? Same comment on others below.